### PR TITLE
MTDSA-18399 parser-validator improvements

### PR DESCRIPTION
**Pull request by mattstephens-hmrc**

### CODE REVIEW

File: test/api/connectors/MockHttpClient.scala [hunk 1]
```diff
@@ -43,47 +43,50 @@ trait MockHttpClient extends MockFactory {
...
+            }
         })
     }
 
     def post[I, T](url: String,
                    config: HeaderCarrier.Config,
                    body: I,
-                   requiredHeaders: Seq[(String, String)] = Seq.empty,
-                   excludedHeaders: Seq[(String, String)] = Seq.empty): CallHandler[Future[T]] = {
+                   requiredHeaders: Seq[(String, String)] = Nil,
```
Comment (SamuelJacobsHMRC) [inline test/api/connectors/MockHttpClient.scala:60]: Just out of curiosity - why the preference of Nil over Seq.empty?

File: test/config/MockAppConfig.scala [hunk 2]
```diff
@@ -38,13 +38,16 @@ trait MockAppConfig extends MockFactory {
...
+    // API Config
     def featureSwitches: CallHandler[Configuration] = (() => mockAppConfig.featureSwitches).expects()
 
     def apiGatewayContext: CallHandler[String] = (() => mockAppConfig.apiGatewayContext).expects()
 
-    def apiStatus(status: String): CallHandler[String] = (mockAppConfig.apiStatus: String => String).expects(status)
+    def apiStatus(version: Version): CallHandler[String] = (mockAppConfig.apiStatus: Version => String).expects(version)
 
-    def endpointsEnabled(version: String): CallHandler[Boolean] = (mockAppConfig.endpointsEnabled: String => Boolean).expects(version)
+    def isApiDeprecated(version: Version): CallHandler[Boolean] = (mockAppConfig.isApiDeprecated: Version => Boolean).expects(version)
```
Comment (SamuelJacobsHMRC) [inline test/config/MockAppConfig.scala:48]: can remove this

File: test/v1/controllers/RetrieveItsaStatusControllerSpec.scala [hunk 3]
```diff
@@ -17,42 +17,34 @@
...
 class RetrieveItsaStatusControllerSpec
-  extends ControllerBaseSpec
+    extends ControllerBaseSpec
     with ControllerTestRunner
     with MockRetrieveItsaStatusService
-    with MockHateoasFactory
-    with MockRetrieveItsaStatusValidator
-    with HateoasLinks {
+    with MockRetrieveItsaStatusValidatorFactory
+    with MockAppConfig {
```
Comment (SamuelJacobsHMRC) [inline test/v1/controllers/RetrieveItsaStatusControllerSpec.scala:41]: should be able to remove with once all the deprecation header related logic has been removed

File: test/v1/controllers/validators/RetrieveItsaStatusValidatorFactorySpec.scala [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+      "passed an invalid nino" in {
+        val result: Either[ErrorWrapper, RetrieveItsaStatusRequestData] =
+          validator(invalidNino, validTaxYear, None, None).validateAndWrapResult()
+
+        result shouldBe Left(
+          ErrorWrapper(correlationId, NinoFormatError)
+        )
+      }
+    }
+
```
Comment (SamuelJacobsHMRC) [inline test/v1/controllers/validators/RetrieveItsaStatusValidatorFactorySpec.scala:74]: should really have test cases for all the error cases i.e. TaxYearFormatError, FutureYearsFormatError, HistoryFormatError to confirm all validations are being done as currently only got NinoFormatError

**Review by SamuelJacobsHMRC — CHANGES_REQUESTED** [review]

File: test/api/connectors/MockHttpClient.scala (hunk 1 above)
Comment (mattstephens-hmrc) [reply test/api/connectors/MockHttpClient.scala:60]: It's shorter

File: test/v1/controllers/validators/RetrieveItsaStatusValidatorFactorySpec.scala (hunk 4 above)
Comment (mattstephens-hmrc) [reply test/v1/controllers/validators/RetrieveItsaStatusValidatorFactorySpec.scala:74]: they're all tested further down in the "multiple errors" test

**Review by SamuelJacobsHMRC — APPROVED** [review]

**Pull request merged**